UI/CombatReport/CombatReportWnd.cpp

```diff
@@ -10,12 +11,34 @@
 #include "CombatLogWnd.h"
 
 #include <GG/TabWnd.h>
+#include <GG/Layout.h>
 
 namespace {
     GG::X COMBAT_LOG_WIDTH(400);
     GG::Y COMBAT_LOG_HEIGHT(300);
-    GG::X RESIZE_MARGIN_X(4);
-    GG::Y RESIZE_MARGIN_Y(4);
+
+    // Duplicated from MapWnd.cpp - used here to find maximum size of the
+    // combat report window.
+    GG::X AppWidth() {
+        if(HumanClientApp* app = HumanClientApp::GetApp())
+            return app->AppWidth();
+        return GG::X0;
+    }
+
+    GG::Y AppHeight() {
+        if(HumanClientApp* app = HumanClientApp::GetApp())
+            return app->AppHeight();
+        return GG::Y0;
+    }
+
+    GG::Pt LimitToAppSize(const GG::Pt& sz_in) {
```

Comment:
It would be better to constrain the Wnd to be within its parent's client area, if it has a parent Wnd. This would eliminate all this out-of-place looking special-case code.